ui/src/components/admin/ArrayToExcelButton.js

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from "react";
+import {
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Select,
+  Checkbox,
+} from "@chakra-ui/react";
+import { DownloadIcon } from "@chakra-ui/icons";
+import { arrayToExcel } from "../helpers/arrayToExcel";
+import cloneDeep from "lodash.clonedeep";
+import { useDispatch } from "react-redux";
+
+const ArrayToExcelButton = ({ apiArray, fileName, buttonTitle }) => {
+  const [showDownloadModal, setShowDownloadModal] = useState(false);
+  const [columnsType, setColumnsType] = useState("1");
+  const [selectedColumns, setSelectedColumns] = useState([]);
+  const dispatch = useDispatch();
+
+  const updateSelectedColumns = (e, column) => {
+    if (e.target.checked) {
+      setSelectedColumns([...selectedColumns, column]);
+    } else {
+      setSelectedColumns(selectedColumns.filter((value) => value !== column));
+    }
+  };
+
+  const apiArrayToExcel = () => {
+    if (columnsType === "1") {
+      arrayToExcel.convertArrayToTable(apiArray, fileName);
+    } else {
+      const customArray = cloneDeep(apiArray);
```

Comment:
not sure if we need a library for this. deep copy can be achieved pretty easily using `JSON.parse(JSON.stringify(apiArray))`